src/main/java/com/puppycrawl/tools/checkstyle/api/CheckstyleAST.java

```diff
@@ -0,0 +1,25 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+/** Base interface used for all ASTs of Checkstyle. */
+public interface CheckstyleAST {
```

Comment:
XPath knows about the underlying AST, so to be able to provide this method without just using `Object` this class was created and all ASTs implement it.

I was originally planning for this to be just `AST` but PMD complained it was too short. I am open to other names.

Right now it is completely empty, but it will have a future purpose for other common areas that fully rely on the specific ASTs provided. The next target area will be the `AstTreeStringPrinter` and making the code agnostic that it doesn't care what AST is given to it.

Future planned methods here will be: print, get line, get column, get type. Some of this when it comes will be breaking changes.

Reply:
I would prefer to not do `Checkstyle’ , what about  `AbstractSyntaxTree`?

Reply:
I have no complaints.